### PR TITLE
43 add post processing interaction output files

### PR DESCRIPTION
**Pull request by rprat-pro**
New output files containing interaction data. 

- An output frequency feature is included directly into `hooke_polyhedron` and `hooke_sphere`
- A new post-processings script `interaction_summary` is added for users in `exaDEM/scripts/post_processing/`
- An associated documentation has been added to doc_exaDEM part 10.3.4 .
- An example using `analysis_interaction_dump_frequency` in `example/polyhedra/analyses/`

Output files example: 

[types.pdf](https://github.com/user-attachments/files/16358615/types.pdf)

[count.pdf](https://github.com/user-attachments/files/16358618/count.pdf)

### CODE REVIEW

**Comment by rprat-pro** [comment]
The list of interaction types is available section 4.3 for polyhedra and  5.1 for spheres.

**Comment by rprat-pro** [comment]
Update example:

 
analyses

count


types

File: src/interaction/include/exaDEM/interaction/classifier.hpp [hunk 1]
```diff
@@ -24,6 +24,24 @@ namespace exaDEM
     ONIKA_HOST_DEVICE_FUNC inline Vec3d& friction(const uint64_t idx) { return m_data[idx].friction; }
   };
 
+
```
Comment (lamarsid) [inline src/interaction/include/exaDEM/interaction/classifier.hpp:27]: add deflection (overlap) d_n to the list

File: src/interaction/include/exaDEM/interaction/classifier.hpp [hunk 2]
```diff
@@ -81,6 +100,18 @@ namespace exaDEM
       return {data_ptr, data_size};
     }
 
+    std::tuple<Vec3d*,Vec3d*,Vec3d*> buffer_p(int id)
+    {
+      auto& analysis = buffers[id]; 
+			// fit size if needed
+      const int size = onika::cuda::vector_size( waves[id] );
+      analysis.resize(size);
+      Vec3d* const cop = onika::cuda::vector_data( analysis.co );
```
Comment (lamarsid) [inline src/interaction/include/exaDEM/interaction/classifier.hpp:109]: rename the co variable to cp or c_pos

File: src/interaction/include/exaDEM/interaction/classifier_analyses.hpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      {
+        auto [i_ptr, size] = ic.get_info(i);
+        auto [co_ptr, fn_ptr, ft_ptr] = ic.buffer_p(i); 
+
+        for(size_t idx = 0 ; idx < size ; idx++)
+        {
+          Vec3d fn = fn_ptr[idx];
+          Vec3d ft = ft_ptr[idx];
+          /** filter empty interactions */
+          if(fn != Vec3d{0,0,0} || ft != Vec3d{0,0,0})
```
Comment (lamarsid) [inline src/interaction/include/exaDEM/interaction/classifier_analyses.hpp:44]: now you can just use the d_n variable to test if the interaction is active or not

**Review by lamarsid — COMMENTED** [review]
some modifications to take account

File: src/interaction/include/exaDEM/interaction/classifier.hpp (hunk 1 above)
Comment (rprat-pro) [reply src/interaction/include/exaDEM/interaction/classifier.hpp:27]: done

File: src/interaction/include/exaDEM/interaction/classifier_analyses.hpp (hunk 3 above)
Comment (rprat-pro) [reply src/interaction/include/exaDEM/interaction/classifier_analyses.hpp]: done

File: src/interaction/include/exaDEM/interaction/classifier_analyses.hpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          {
+            auto& I = i_ptr[idx];
+            /** Note that an interaction between two particles present on two sub-domains should not be counted twice. */
+            if(filter_duplicates(grid, I))
+            {
+              stream << I.id_i  << "," << I.id_j << ","; 
+              stream << I.sub_i << "," << I.sub_j << ","; 
+              stream << I.type << ","; 
+              stream << co_ptr[idx] << ",";
```
[the remark this answers is not in the document]
Comment (rprat-pro) [reply src/interaction/include/exaDEM/interaction/classifier_analyses.hpp]: done

File: src/interaction/include/exaDEM/interaction/classifier.hpp (hunk 2 above)
Comment (rprat-pro) [reply src/interaction/include/exaDEM/interaction/classifier.hpp]: done

**Review by lamarsid — APPROVED** [review]
bravo

**Pull request merged**